src/AstrometryFit.cc

```diff
@@ -422,75 +300,55 @@ void AstrometryFit::LSDerivatives2(const FittedStarList &fsl, TripletList &tList
         with the measurement terms. Since P(fs) = 0, we have: */
         res[0] = -rsProj.x;
         res[1] = -rsProj.y;
-        halpha = h * alpha;
-        // grad = h*w*res
-        hw = h * w;
-        grad = hw * res;
-        // now feed in triplets and rhs
+        halpha = H * alpha;
+        // grad = H*W*res
+        HW = H * W;
+        grad = HW * res;
+        // now feed in triplets and fullGrad
         for (unsigned ipar = 0; ipar < npar_tot; ++ipar) {
             for (unsigned ic = 0; ic < 2; ++ic) {
                 double val = halpha(ipar, ic);
                 if (val == 0) continue;
-#if (TRIPLET_INTERNAL_COORD == COL)
-                tList.addTriplet(indices[ipar], kTriplets + ic, val);
-#else
-                tList.addTriplet(kTriplets + ic, indices[ipar], val);
-#endif
+                tripletList.addTriplet(indices[ipar], kTriplets + ic, val);
             }
-            rhs(indices[ipar]) += grad(ipar);
+            fullGrad(indices[ipar]) += grad(ipar);
         }
         kTriplets += 2;  // each measurement contributes 2 columns in the Jacobian
     }
-    tList.setNextFreeIndex(kTriplets);
-}
-
-//! this routine computes the derivatives of all LS terms, including the ones that refer to references stars,
-//! if any
-void AstrometryFit::LSDerivatives(TripletList &tList, Eigen::VectorXd &rhs) const {
-    auto L = _assoc.getCcdImageList();
-    for (auto const &im : L) {
-        LSDerivatives1(*im, tList, rhs);
-    }
-    LSDerivatives2(_assoc.fittedStarList, tList, rhs);
+    tripletList.setNextFreeIndex(kTriplets);
 }
 
-// This is almost a selection of lines of LSDerivatives1(CcdImage ...)
-/* This routine (and the following one) is template because it is used
-both with its first argument as "const CCdImage &" and "CcdImage &",
-and I did not want to replicate it.  The constness of the iterators is
-automagically set by declaring them as "auto" */
-
-template <class ImType, class Accum>
-void AstrometryFit::accumulateStatImage(ImType &image, Accum &accu) const {
+// This is almost a selection of lines of leastSquareDerivativesMeasurement(CcdImage ...)
+void AstrometryFit::accumulateStatImage(CcdImage const &ccdImage, Chi2Accumulator &accum) const {
     /**********************************************************************/
-    /**  Changes in this routine should be reflected into LSDerivatives1  */
+    /**  Changes in this routine should be reflected into leastSquareDerivativesMeasurement  */
```

Comment:
There's math in both places that has to remain consistent, but can't be easily refactored out. Suggestions on how to make that fact clearer are welcome.